### PR TITLE
[CDAP-15787] Updated XMLToJSON for new Validation API

### PR DESCRIPTION
**Pull request by dli357**
Added two unit tests as well for the new validation checks.

### CODE REVIEW

File: transform-plugins/src/main/java/io/cdap/plugin/XMLToJSON.java [hunk 1]
```diff
@@ -80,14 +83,16 @@ public void initialize(TransformContext context) throws Exception {
...
 
     Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
     if (inputSchema != null) {
       if (inputSchema.getField(config.inputField) == null) {
-        throw new IllegalArgumentException(String.format("Field %s is not present in input schema", config.inputField));
-      }
-      if (inputSchema.getField(config.inputField).getSchema().getType() != Schema.Type.STRING) {
-        throw new IllegalArgumentException(String.format("Field %s must be of type string.", config.inputField));
+        collector.addFailure(String.format("Field '%s' must be present in input schema.", config.inputField), null)
+            .withConfigProperty(INPUT_FIELD);
```
Comment (albertshau) [inline transform-plugins/src/main/java/io/cdap/plugin/XMLToJSON.java]: indentation (and rest of PR)

File: transform-plugins/src/main/java/io/cdap/plugin/XMLToJSON.java [hunk 2]
```diff
@@ -80,14 +83,16 @@ public void initialize(TransformContext context) throws Exception {
...
+      } else if (inputSchema.getField(config.inputField).getSchema().getType() != Schema.Type.STRING) {
```
Comment (albertshau) [inline transform-plugins/src/main/java/io/cdap/plugin/XMLToJSON.java]: this is overly verbose, it's easier to read if it's like

```
Schema.Field field = inputSchema.getField(config.inputField);
if (field == null) {
  ...
} else if (...
```

File: transform-plugins/src/main/java/io/cdap/plugin/XMLToJSON.java (hunk 1 above)
Comment (dli357) [reply transform-plugins/src/main/java/io/cdap/plugin/XMLToJSON.java]: Fixed

File: transform-plugins/src/main/java/io/cdap/plugin/XMLToJSON.java (hunk 2 above)
Comment (dli357) [reply transform-plugins/src/main/java/io/cdap/plugin/XMLToJSON.java]: Fixed

**Review by albertshau — APPROVED** [review]

**Pull request merged**